PictureApp.API/Controllers/UsersController .cs

```diff
@@ -15,25 +15,73 @@ public class UsersController : ControllerBase
     {
         private readonly IUserService _userService;
         private readonly IMapper _mapper;
+        private readonly IPhotoService _photoService;
 
-        public UsersController(IUserService userService, IMapper mapper)
+        public UsersController(IUserService userService, IMapper mapper, IPhotoService photoService)
         {
             _userService = userService ?? throw new ArgumentNullException(nameof(userService));
             _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _photoService = photoService;
         }
 
         [HttpGet("{id}")]
-        public IActionResult GetUser(int id)
+        public async Task<IActionResult> GetUser(int id)
         {
-            return Ok(_userService.GetUser(id));
+            try
+            {
+                CheckIfCurrentUserIsCorrect(id);
```

Comment:
This check will raise an exception for any id that is not from current user. Is it our purpose?